File: algorithm/golang/two_sum_test.go
package algo_test

import (
	"testing"

	"hungknow.com/algo"
)

func TestTwoSum(t *testing.T) {
	nums := []int{2, 7, 11, 15}
	target := 9
	expected := []int{0, 1}

	result := algo.TwoSum(nums, target)

	if len(result) != len(expected) {
		t.Errorf("Expected length of result to be %d, but got %d", len(expected), len(result))
	}

	for i := 0; i < len(result); i++ {
		if result[i] != expected[i] {
			t.Errorf("Expected result[%d] to be %d, but got %d", i, expected[i], result[i])
		}
	}
}

func TestTwoSum_EmptyInput(t *testing.T) {
	nums := []int{}
	target := 9
	expected := []int{}

	result := algo.TwoSum(nums, target)

	if len(result) != len(expected) {
		t.Errorf("Expected length of result to be %d, but got %d", len(expected), len(result))
	}
}

func TestTwoSum_NoSolution(t *testing.T) {
	nums := []int{2, 7, 11, 15}
	target := 100
	expected := []int{}

	result := algo.TwoSum(nums, target)

	if len(result) != len(expected) {
		t.Errorf("Expected length of result to be %d, but got %d", len(expected), len(result))
	}

}

File: algorithm/kotlin/lib/src/test/java/com/hungknow/TwoSumTest.java
package com.hungknow;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class TwoSumTest {

    @Test
    public void testTwoSum() {
        TwoSum twoSum = new TwoSum();
        int[] nums = { 2, 7, 11, 15 };
        int target = 9;
        int[] expected = { 0, 1 };
        int[] result = twoSum.twoSum(nums, target);
        Assertions.assertArrayEquals(expected, result);
    }

    @Test
    public void testTwoSum_NoSolution() {
        TwoSum twoSum = new TwoSum();
        int[] nums = { 2, 7, 11, 15 };
        int target = 0;
        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            twoSum.twoSum(nums, target);
        });
    }
}
